File: internetbanking/src/app/services/lichsugiaodich.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class LichsugiaodichService {

  constructor(private http: HttpClient) { }

  getLichSuGiaoDich(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getLichSuGiaoDich`, value);
  }

  getChuyenTien(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getChuyenTien`, value);
  }

  getNhanTien(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getNhanTien`, value);
  }

  getTraNo(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getTraNo`, value);
  }

  getNguoiKhacTraNo(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getNguoiKhacTraNo`, value);
  }

  getLichSuGiaoDichByEmployee(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getLichSuGiaoDichByEmployee`, value);
  }

  getLichSuGiaoDichByEmployeeDetails(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getLichSuGiaoDichByEmployeeDetails`, value);
  }

  getLichSuGiaoDichByAdmin(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getLichSuGiaoDichByAdmin`, value);
  }

  getLichSuGiaoDichByAdminDetails(value: any) {
    return this.http.post(`${environment.apiUrl}/lichsugiaodich/getLichSuGiaoDichByAdminDetails`, value);
  }
}

File: KhoaBanking/src/main/java/com/khoa/service/LichSuGiaoDichService.java
package com.khoa.service;

import java.util.List;

import com.khoa.dto.DanhSachLichSuGiaoDichDTO;
import com.khoa.dto.LichSuGiaoDichDetailsDTO;
import com.khoa.entity.LichSuGiaoDich;


public interface LichSuGiaoDichService {
	List<LichSuGiaoDich> findAll();
	LichSuGiaoDich findById(int id);
	void add(LichSuGiaoDich dto);
	void update(LichSuGiaoDich dto);
	void delete(int id);
	DanhSachLichSuGiaoDichDTO getLichSuGiaoDich(int manguoigui);
	DanhSachLichSuGiaoDichDTO getChuyenTien(DanhSachLichSuGiaoDichDTO danhSachLichSuGiaoDichDTO);
	DanhSachLichSuGiaoDichDTO getNhanTien(DanhSachLichSuGiaoDichDTO danhSachLichSuGiaoDichDTO);
	DanhSachLichSuGiaoDichDTO getTraNo(DanhSachLichSuGiaoDichDTO danhSachLichSuGiaoDichDTO);
	DanhSachLichSuGiaoDichDTO getNguoiKhacTraNo(DanhSachLichSuGiaoDichDTO danhSachLichSuGiaoDichDTO);
	List<LichSuGiaoDich> getLichSuGiaoDichByEmployee(DanhSachLichSuGiaoDichDTO danhSachLichSuGiaoDichDTO);
	LichSuGiaoDichDetailsDTO getLichSuGiaoDichByEmployeeDetails(int id);
}
